src/java/org/apache/cassandra/io/compress/CompressionMetadata.java

```diff
@@ -540,6 +554,7 @@ public void resetAndTruncate(int chunkIndex)
 
         protected Throwable doPostCleanup(Throwable failed)
         {
+            NATIVE_MEMORY_USAGE.addAndGet(-offsets.size());
```

Comment:
If we have taken a shared copy in `open` above (which we usually do), the memory `offsets` holds will not be released by `close` below.